internal/tuplemapper/tuplemapper.go

```diff
@@ -0,0 +1,56 @@
+package tuplemapper
+
+import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/tuple"
+)
+
+type MapperKind int64
+
+const (
+	// NestedUsersetKind is a mapper that returns the userset ID from the tuple's user field.
+	NestedUsersetKind MapperKind = iota
+	NestedTTUKind
+)
+
+type Mapper interface {
+	storage.Iterator[string]
+}
+
+type NestedUsersetMapper struct {
+	Iter storage.TupleKeyIterator
+}
+
+func (n NestedUsersetMapper) Next(ctx context.Context) (string, error) {
+	tupleRes, err := n.Iter.Next(ctx)
+	if err != nil {
+		return "", err
+	}
+	return n.Map(tupleRes)
+}
+
+func (n NestedUsersetMapper) Stop() {
+	n.Iter.Stop()
+}
+
+func (n NestedUsersetMapper) Head(ctx context.Context) (string, error) {
+	tupleRes, err := n.Iter.Head(ctx)
+	if err != nil {
+		return "", err
+	}
+	return n.Map(tupleRes)
+}
+
+func (n NestedUsersetMapper) Map(t *openfgav1.TupleKey) (string, error) {
```

Comment:
nitpik: 
1. Why does this need to be a function of NestedUsersetMapper instead a function that does not belong to any struct?
2. why is this public?

Reply:
1. The idea is that in the future there will be a NestedTTUMapper that will do this: `return t.GetUser(), nil`
2. No reason, made private now :)